### ISSUE TITLE
MOE models that based on Gemma cannot work

### ISSUE THREAD

**Issue by PersonaTalk** [arcee-ai/mergekit#352]
I use mergekit-moe to generate MOE model with several same Gemma models(mode hidden), but the result model output meaningless results like in https://github.com/arcee-ai/mergekit/issues/218#issuecomment-2027402773.  It didn't happened when I merged models that based on Llama.

**Comment by pharaohcaptain** [comment]
Is gemma's moe not supported yet?

**Comment by pharaohcaptain** [comment]
I merge successfully. But two identical models theoretically output the same result as the original model. Here I output something like this: "milf milf milf milf milf milf milf milf milf milf milf milf milf milf increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa increa".
The llama model does not suffer from these problems.
Can anyone help me?

**Comment by femiariel** [comment]
Yeah, unfortunately, for the moment , only mixtral, deepseek and Qwen-MoE architecture for the mixture of Experts are supported.
Gemma's architecture isn't supported for the moment, you can test with the Framework mergoo.
